sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -959,6 +959,37 @@ class CollationStringExpressionsSuite
     assert(collationMismatch.getErrorClass === "DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE")
   }
 
+  test("DateFormat string expression with collation") {
+    case class DateFormatTestCase[R](d: String, f: String, c: String, r: R)
```

Comment:
can we maybe use more expressive names instead of single letters?

Reply:
+1 on this

@nebojsa-db you'll notice that naming in CollationSQLExpressionsSuite is a bit better too